src/bootchooser.c

```diff
@@ -62,7 +62,7 @@ static gboolean barebox_state_get(const gchar* bootname, BareboxSlotState *bb_st
 	g_autoptr(GSubprocess) sub = NULL;
 	GError *ierror = NULL;
 	GInputStream *instream;
-	GDataInputStream *datainstream;
+	g_autoptr(GDataInputStream) datainstream;
```

Comment:
Must be initialized to `NULL`. Otherwise, a non-initialized value will be destructed whenever `g_subprocess_newv` fails.

Reply:
You're right, missed that. Thx!